### PR TITLE
feat: Add outputs for the full list of subnets created and their attributes

### PR DESCRIPTION
**Pull request by certara-mchamberland**
## Description
This adds outputs mapping subnet IDs to AZs. This allows comparing AZs with a remote PrivateLink Endpoint Service to find out in what subnet we can create an endpoint locally, as PrivateLink requires both ends use the same AZs.

## Breaking Changes
No, these are only new outputs.

## How Has This Been Tested?
- [ ] I have updated at least one of the `examples/*` to demonstrate and validate my change(s) - Not sure its relevant for outputs
- [X] I have tested and validated these changes using one or more of the provided `examples/*` projects - I applied it directly to a module of mine that, among other things, locates matching AZs on both ends of a PrivateLink VPC Endpoint

- [X] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

**Comment by certara-jgitlin** [comment]
LGTM! 👍

**Comment by certara-mchamberland** [comment]
Could a maintainer pls take a look at this? I'm happy to keep working on it if its not currently adequate. Thanks!

File: outputs.tf
Comment (bryantbiggs) [inline outputs.tf:160]: rather than this custom mapping, I think a better approach would be to just output all attributes of the subnets created and let users use those as they please (you can use this to create this mapping for example in your own implementation)

```hcl
output "public_subnets" {
  description = "List of the subnets created and their associate attributes"
  value       = aws_subnet.public
}
```

**Review by bryantbiggs — CHANGES_REQUESTED** [review]

File: outputs.tf
Comment (certara-mchamberland) [reply outputs.tf:160]: I'd built it this way to avoid a breaking change, hoping for a fast release, but I can arrange this too. Right now my org is working from my fork anyways. Alternatively, I could submit a second PR implementing the "breaking change" for merge in the next major version, your call!

File: outputs.tf
Comment (bryantbiggs) [inline outputs.tf:110]: No need for the `[*]`, they can all be dropped if they are the last attribute selector
```suggestion
  value       = aws_subnet.public
```

**Pull request merged**
